src/ci/docker/scripts/musl-toolchain.sh

```diff
@@ -45,11 +45,11 @@ export CFLAGS="-fPIC -g1 $CFLAGS"
 
 git clone https://github.com/richfelker/musl-cross-make # -b v0.9.9
 cd musl-cross-make
-# A version that includes support for building musl 1.2.3
-git checkout fe915821b652a7fa37b34a596f47d8e20bc72338
+# A version that includes support for building musl 1.2.5
+git checkout e149c31c48b4f4a4c9349ddf7bc0027b90245afc
```

Comment:
I think we should bump to [7b4c7b315226835bab03da73a45945acb1b3bedf](https://github.com/richfelker/musl-cross-make/commit/7b4c7b315226835bab03da73a45945acb1b3bedf) or higher to pick up the patches for CVE-2025-26519